dataclasses_json/api.py

```diff
@@ -25,11 +29,181 @@ class LetterCase(Enum):
     PASCAL = pascalcase
 
 
+class IgnoreUndefinedParameters(UndefinedParameterAction):
+    """
+    This action does nothing when it encounters undefined parameters.
+    The undefined parameters can not be retrieved after the class has been created.
+    """
+
+    @staticmethod
+    def handle_from_dict(cls, kvs: Dict) -> Dict[str, Any]:
+        known_given_parameters, _ = UndefinedParameterAction._separate_defined_undefined_kvs(cls=cls, kvs=kvs)
+        return known_given_parameters
+
+    @staticmethod
+    def create_init(obj) -> Callable:
+        original_init = obj.__init__
+        init_signature = inspect.signature(original_init)
+
+        @functools.wraps(obj.__init__)
+        def _ignore_init(self, *args, **kwargs):
+            known_kwargs, _ = CatchAllUndefinedParameters._separate_defined_undefined_kvs(obj, kwargs)
+            num_params_takeable = len(init_signature.parameters) - 1  # don't count self
+            num_args_takeable = num_params_takeable - len(known_kwargs)
+
+            args = args[:num_args_takeable]
+            bound_parameters = init_signature.bind_partial(self, *args, **known_kwargs)
+            bound_parameters.apply_defaults()
+
+            arguments = bound_parameters.arguments
+            arguments.pop("self", None)
+            final_parameters = IgnoreUndefinedParameters.handle_from_dict(obj, arguments)
+            original_init(self, **final_parameters)
+
+        return _ignore_init
+
+
+class RaiseUndefinedParameters(UndefinedParameterAction):
+    """
+    This action raises UndefinedParameterError if it encounters an undefined parameter during initialization.
+    """
+
+    @staticmethod
+    def handle_from_dict(cls, kvs: Dict) -> Dict[str, Any]:
+        known, unknown = UndefinedParameterAction._separate_defined_undefined_kvs(cls=cls, kvs=kvs)
+        if len(unknown) > 0:
+            raise UndefinedParameterError(f"Received undefined initialization arguments {unknown}")
+        return known
+
+
+CatchAll = Optional[CatchAllVar]
+
+
+class CatchAllUndefinedParameters(UndefinedParameterAction):
+    """
+    This class allows to add a field of type utils.CatchAll which acts as a dictionary into which all
+    undefined parameters will be written.
+    These parameters are not affected by LetterCase.
+    If no undefined parameters are given, this dictionary will be empty.
+    """
+
+    @staticmethod
+    def handle_from_dict(cls, kvs: Dict) -> Dict[str, Any]:
+        known, unknown = UndefinedParameterAction._separate_defined_undefined_kvs(cls=cls, kvs=kvs)
```

Comment:
this function is very long, would like to see it split into private smaller methods

Reply:
Moved computation of default value to its own private method